src/components/clusterprovider/minikube/minikube.ts

```diff
@@ -198,13 +198,42 @@ async function stopMinikube(context: Context): Promise<void> {
     });
 }
 
+function getStatusFormatString(currentVersion: string): string {
+    if (currentVersion.startsWith("v0.")) {
+        if (parseInt(currentVersion.split(".")[1]) < 31) {
+            // minikube < 0.31.0 uses this format
+            return `"{{.MinikubeStatus}}","{{.ClusterStatus}}","{{.KubeconfigStatus}}"`;
+        } else {
+            // minikube < 1.0.0 uses this format
+            return `"{{.Host}}",{ "kubelet": "{{.Kubelet}}", "apiserver": "{{.ApiServer}}" },"{{.Kubeconfig}}"`;
+        }
+    }
+
+    // minikube >= 1.0.0 uses this format
+    return `"{{.Host}}",{ "kubelet": "{{.Kubelet}}", "apiserver": "{{.APIServer}}" },"{{.Kubeconfig}}"`;
+}
+
 async function minikubeStatus(context: Context): Promise<MinikubeInfo> {
     if (!await checkPresent(context, CheckPresentMode.Silent)) {
         throw new Error('minikube executable could not be found!');
     }
 
+    const sr = await context.shell.exec(`"${context.binPath}" version`);
+    if (!sr || sr.code !== 0) {
+        throw new Error(`Error checking for minikube version: ${sr ? sr.stderr : 'cannot run minikube'}`);
+    }
+    const lines = sr.stdout.split('\n')
+                           .map((l) => l.trim())
+                           .filter((l) => l.length > 0);
+    if (lines.length !== 1) {
+        throw new Error(`Unexpected output for minikube version check: ${lines}`);
+    }
+    const currentVersion = extractVersion(lines[0]);
```

Comment:
I think we have similar code to this in `minikubeUpgradeAvailable` - is it worth pulling out the common code?

Reply:
yes, i copied the code (for me it was the fastest way; sometimes i'm lazy like this), but i will look into pulling out the common code